src/components/Api.js

```diff
@@ -0,0 +1,179 @@
+export class Api {
+  constructor(config) {
+    this.url = config.baseUrl;
+    this.headers = config.headers;
+  }
+
+  getInitialCards() {
+    return fetch(`${this.url}/cards`, {
+      headers: {
+        authorization: "fe7e07a4-81c5-490b-807b-e6a7cec619a0",
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      // if the server returns an error, reject the promise
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  getUserInfo() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      method: "GET",
+      headers: {
+        authorization: "fe7e07a4-81c5-490b-807b-e6a7cec619a0",
+        "Content-Type": "application/json",
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  updateUserProfile(userData) {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      method: "PATCH",
+      headers: {
+        authorization: "fe7e07a4-81c5-490b-807b-e6a7cec619a0",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(userData),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
```

Comment:
Needs correction: Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```

Reply:
I removed the catch blocks from the API file. All catch blocks are in index.js where needed.